src/init.cpp

```diff
@@ -1807,6 +1814,25 @@ bool AppInitMain(NodeContext& node)
         return false;
     }
 
+    // Read asmap file if configured
+    if (gArgs.IsArgSet("-asmap")) {
+        std::string asmap_file = gArgs.GetArg("-asmap", "");
+        if (asmap_file.empty()) {
+            asmap_file = DEFAULT_ASMAP_FILENAME;
+        }
+        const fs::path asmap_path = GetDataDir() / asmap_file;
```

Comment:
This implies that the asmap file will always be in the default data directory. Should it allow for absolute paths as well? For example, I ended up with this error on my first naive run of this branch:

```
$ bitcoind -testnet -asmap=/Users/matthewzipkin/Desktop/work/asmap/demo.map
...
Error: Could not find or parse specified asmap:
'"/Users/matthewzipkin/Library/Application Support/Bitcoin/testnet3/Users/matthewzipkin/Desktop/work/asmap/demo.map"'
```

Reply:
At this point I'm not even sure what should be the right behaviour here. summoning @jonatack.

Reply:
@pinheadmz @naumenkogs Yes, the PR as authored specifies a relative path and I tested it as such:
```
gArgs.AddArg("-asmap=<file>", "Specify asn mapping, relative to the -datadir path", ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```
I agree allowing for absolute paths would be handy. Will perhaps propose a pair of commits to do this and add tests for expected behavior as per https://github.com/bitcoin/bitcoin/pull/16702#discussion_r357139904 for relative and absolute paths, that can be pulled in or proposed separately, per @naumenkogs' preference, so he doesn't have to bother with it.

Reply:
Of course, specifying an asmap file outside of the datadir is currently possible:

` bitcoind -asmap=../projects/bitcoin/asmap/demo.map`

```
2019-12-25T11:13:05Z Opened asmap file "/home/jon/.bitcoin/../projects/bitcoin/asmap/demo.map" (932999 bytes) from disk.
2019-12-25T11:13:05Z Using asmap version 7ead927689f884bc4021618017cc1438d9328578779edd2adfbed044fb52928e for IP bucketing.
```

Reply:
done in #17812